sigstore/dsse.py

```diff
@@ -210,6 +223,26 @@ def _from_json(cls, contents: bytes | str) -> Envelope:
         inner = _Envelope().from_json(contents)
         return cls(inner)
 
+    @classmethod
+    def _from_payload(
+        cls, payload: RawPayload, sigs: list[Signature]
+    ) -> Envelope:
+        """Return an unsigned DSSE envelope.
+
+        Args:
+            payload_type (str): The envelope's payload type
+            payload (bytes): The envelope's payload
+
+        Returns:
+            Envelope: An unsigned DSSE envelope
+        """
+        inner = _Envelope(
+            payload=payload.data,
+            payload_type=payload.type,
+            signatures=sigs,
+        )
+        return cls(inner)
```

Comment:
Merging `_sign_payload` with `_sign` should eliminate the need for this new helper 🙂